### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by igoradamenko**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
 
-This gem will help you to validate and synchronize your API in strict accordance to the documentation in
+This gem helps you to validate and synchronize your API in strict accordance to the documentation in
 [API Blueprint](https://apiblueprint.org/) format.
-To do this it automatically searches received requestes and responses in the documentation and run validates
-json-schemas.
+To do this it automatically searches received requests and responses in the documentation and run
+JSON-schemas validators.
+
+## Contents
```
Comment (igoradamenko) [inline README.md:11]: README includes module documentation, so I thought that ToC might be useful. But not sure.

File: README.md [hunk 2]
```diff
@@ -20,107 +41,189 @@ Add this line to your application's Gemfile:
...
-From him the `Esplanade::Request::PrefixNotMatch`, `Esplanade::Request::NotDocumented`, `Esplanade::Request::BodyIsNotJson` and `Esplanade::Request::Invalid` are inherited.
+Parent class for those described below. Inherited from `Esplanade::Error`.
 
 #### Esplanade::Request::PrefixNotMatch
 
-Error message: `{:method=>"method", :path=>"path", :content_type=>"content_type"}`.
+Error message format: 
+
+```ruby
+{
```
Comment (igoradamenko) [inline README.md:102]: I'm not sure that such “beautification” of Ruby maps is alright in Ruby world. Totally ready to revert the changes and make these maps one-lined.

File: README.md [hunk 3]
```diff
@@ -20,107 +41,189 @@ Add this line to your application's Gemfile:
...
+    :method => "method",
+    :path => "path"
+  },
+  :status => "status"
+}
+```
 
 #### Esplanade::Response::BodyIsNotJson
 
 Only if the documentation for all the responses of one request indicates that `Content-Type: application/json`.
```
Comment (igoradamenko) [inline README.md:180]: I couldn't get the meaning behind this line, and it means it should be rewritten.

Does it mean that `Esplanade::Response::BodyIsNotJson` is thrown only when request body isn't JSON and there's no non-JSON responses documented for the endpoint?

**Comment by igoradamenko** [comment]
@tuwilof, is it fine that one of the GitHub Actions tries to publish gem and it can't?

```
You have enabled multi-factor authentication. Please enter OTP code.
You have enabled multifactor authentication but no OTP code provided. Please fill it and retry.
##[error]Process completed with exit code 1.
```

https://github.com/funbox/esplanade/pull/27/checks?check_run_id=875006959

File: README.md (hunk 1 above)
Comment (tuwilof) [reply README.md:11]: looks interesting, let it be

Comment (tuwilof) [reply README.md:11]: it's okay, let it be

File: README.md (hunk 2 above)
Comment (tuwilof) [reply README.md:102]: it's okay, let it be

File: README.md (hunk 3 above)
Comment (tuwilof) [reply README.md:180]: yep

Comment (igoradamenko) [reply README.md:180]: Okay, give me a minute to rewrite it.

Comment (igoradamenko) [reply README.md]: Done.
